libr/core/cmd_zign.c

```diff
@@ -983,43 +983,74 @@ TODO: add useXRefs, useName
 	return retval;
 }
 
+static void print_possible_matches(RList *list) {
+	RListIter *itr;
+	RSignCloseMatch *row;
+	r_list_foreach (list, itr, row) {
+		// total score
+		if (row->bscore > 0.0 && row->gscore > 0.0) {
+			r_cons_printf ("%02.5lf  ", row->score);
+		}
+		if (row->bscore > 0.0) {
+			r_cons_printf ("%02.5lf B  ", row->bscore);
+		}
+		if (row->gscore > 0.0) {
+			r_cons_printf ("%02.5lf G  ", row->gscore);
+		}
+		r_cons_printf (" %s\n", row->item->name);
+	}
+}
+
 static bool bestmatch(void *data, const char *input) {
 	RCore *core = (RCore *)data;
-	bool found = false;
 	int count = 5;
 
 	if (!R_STR_ISEMPTY (input)) {
 		count = atoi (input);
 		if (count <= 0) {
 			eprintf ("[!!] invalid number %s\n", input);
-			return found;
+			return false;
 		}
 	}
 
 	RAnalFunction *fcn = r_anal_get_fcn_in (core->anal, core->offset, 0);
+	if (!fcn) {
+		eprintf ("No function at 0x%08" PFMT64x "\n", core->offset);
+		return false;
+	}
 	RSignItem *item = r_sign_item_new ();
 	if (!item) {
-		return NULL;
+		return false;
 	}
 
-	if (!r_sign_addto_item (core->anal, item, fcn, R_SIGN_GRAPH)) {
-		r_sign_item_free (item);
-		return NULL;
+	if (r_config_get_i (core->config, "zign.bytes")) {
+		r_sign_addto_item (core->anal, item, fcn, R_SIGN_BYTES);
+		RSignBytes *b = item->bytes;
+		int minsz = r_config_get_i (core->config, "zign.minsz");
```

Comment:
move this one line below

Reply:
I am not sure what you mean here. I can't move `minsz` declaration inside of the `if` body because I use `minsz` in the condition. I could call `r_config_get_i` in the condition and then again in `eprintf` to avoid a variable since it only happens on the unlikely event that the function is too small.

Or were you talking about moving `RSignBytes *b` down below `minsz`?